contracts/toCKB-typescript/src/switch/bonding/mod.rs

```diff
@@ -1,7 +1,94 @@
 use crate::switch::ToCKBCellDataTuple;
-use crate::utils::types::Error;
+use crate::utils::config::COLLATERAL_PERCENT;
+use crate::utils::tools::{get_xchain_kind, XChainKind};
+use crate::utils::types::{Error, ToCKBCellDataView};
+use bech32::{self, FromBase32};
+use ckb_std::ckb_constants::Source;
+use ckb_std::ckb_types::{bytes::Bytes, prelude::*};
+use ckb_std::high_level::{load_cell_capacity, load_witness_args};
 use core::result::Result;
+use molecule::prelude::Vec;
 
-pub fn verify(_toCKB_data_tuple: &ToCKBCellDataTuple) -> Result<(), Error> {
+pub fn verify_data(
+    input_toCKB_data: &ToCKBCellDataView,
+    out_toCKB_data: &ToCKBCellDataView,
+) -> Result<u128, Error> {
+    let coin_kind = get_xchain_kind()?;
+    let amount: u128 = match coin_kind {
+        XChainKind::Btc => {
+            if out_toCKB_data.get_btc_lot_size()? != input_toCKB_data.get_btc_lot_size()? {
+                return Err(Error::InvariantDataMutated);
+            }
+            let (hrp, data) = bech32::decode(
+                core::str::from_utf8(out_toCKB_data.x_lock_address.as_ref()).unwrap(),
+            )
+            .map_err(|_| Error::XChainAddressInvalid)?;
+            if hrp != "bc" {
+                return Err(Error::XChainAddressInvalid);
+            }
+            let raw_data = Vec::<u8>::from_base32(&data).unwrap();
+            if &raw_data[..2] != &[0x00, 0x14] {
+                return Err(Error::XChainAddressInvalid);
+            }
+            if raw_data.len() != 22 {
+                return Err(Error::XChainAddressInvalid);
+            }
```

Comment:
```suggestion
            if raw_data.len() != 22 {
                return Err(Error::XChainAddressInvalid);
            }
            if &raw_data[..2] != &[0x00, 0x14] {
                return Err(Error::XChainAddressInvalid);
            }
```

It may panic here if you don't check the length first.